x-pack/plugins/ml/public/datavisualizer/datavisualizer_controller.js

```diff
@@ -305,12 +305,18 @@ module
 
       const metricCards = [];
 
-      // Add a config for 'document count', identified by no field name.
-      metricCards.push({
-        type: ML_JOB_FIELD_TYPES.NUMBER,
-        existsInDocs: true,
-        loading: true
-      });
+      // Add a config for 'document count', identified by no field name if index is timeseries based
+      if (indexPattern.timeFieldName !== undefined) {
+        metricCards.push({
+          type: ML_JOB_FIELD_TYPES.NUMBER,
+          existsInDocs: true,
+          loading: true
+        });
+      } else {
+        // disable timeRangeSelector and remove sidebar if index not timeseries based
+        timefilter.disableTimeRangeSelector();
```

Comment:
I'm not sure how the overall single page app react to this or remembers that settings, maybe it's necesseary to also explicitly enable it when it's a timeseries based index? (same for `showSidebar` below)

Reply:
`timefilter.enableTimeRangeSelector();` is called when the page loads, so is on by default.
navigating away and back will switch it on again.
but yeah, for the sake of making this obvious, the enabling lines could be moved into this `if` block.

Reply:
Like @jgowdyelastic said, the enabling happens when the controller is first loaded (on line 63). `showSidebar` is also set on initial loading (line 100) so these only get changed if there is no timeFieldName on the index. 

I can go ahead and move the enabling to this if block if that makes it clearer 👍

Reply:
Actually, looks like we need the `timefilter` to be enable on initial load so we can set 
```
$scope.earliest = timefilter.getActiveBounds().min.valueOf();
$scope.latest = timefilter.getActiveBounds().max.valueOf();
```
In the next few lines. I _could_ do a workaround for this but it seems like too much time for just trying to clarify something. So I'll just add a comment.

Reply:
Thanks for clarifying! LGTM as is :)